api/src/main/java/org/apache/iceberg/SortField.java

```diff
@@ -80,7 +80,7 @@ public NullOrder nullOrder() {
    * @return true if this order satisfies the given order
    */
   public boolean satisfies(SortField other) {
-    if (this == other) {
+    if (Objects.equals(this, other)) {
```

Comment:
even though using Ref equlity is correct here, it just takes the reader much longer to verify whether it's the intention or not than just using `equals()` here